src/components/shopForm.tsx

```diff
@@ -0,0 +1,170 @@
+import { Uploader } from "uploader";
+import { UploadDropzone } from "react-uploader";
+import { useFormik } from "formik";
+import { formSchema } from "../utils/validations";
+import { useMutation } from "@tanstack/react-query";
+import toast, { Toaster } from "react-hot-toast";
+
+import axios from "axios";
+import ShopFormUIElements from "./shopFormUIElements";
+import router from "next/router";
+import { useState } from "react";
+
+interface Shop {
+  name: string;
+  email: string;
+  phone: string;
+  description?: string;
+}
+
+export default function ShopForm() {
+  const [textAreaCount, setTextAreaCount] = useState(0);
+  const input =
+    "rounded border border-solid border-black p-2 font-openSans text-sm text-stone-500";
+
+  const uploader = Uploader({
+    // Get production API keys from Upload.io
+    apiKey: "free",
+  });
+
+  const formik = useFormik({
+    initialValues: {
+      name: "",
+      email: "",
+      phone: "",
+      description: " ",
+    },
+
+    validationSchema: formSchema,
+
+    onSubmit: async (values: Shop) => {
```

Comment:
You could've put a check here to see if the mutation is currently loading to prevent double submits. 

https://tanstack.com/query/v4/docs/guides/mutations#:~:text=isLoading%20or%20status%20%3D%3D%3D%20%27loading%27%20%2D%20The%20mutation%20is%20currently%20running

Reply:
ah yeah maybe a:

`  {mutation.isLoading ? (
        'Adding Shop...'
      )`

wouldve helped instead of 

` <button
            disabled={!(formik.isValid && formik.dirty) || formik.isSubmitting}>`